internal/kms/kes.go

```diff
@@ -172,7 +172,11 @@ var _ KMS = (*kesClient)(nil) // compiler check
 // Stat returns the current KES status containing a
 // list of KES endpoints and the default key ID.
 func (c *kesClient) Stat(ctx context.Context) (Status, error) {
-	if _, err := c.client.Version(ctx); err != nil {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
```

Comment:
Is this wide lock really necessary ?

Reply:
Look at other code, this was actually a bug this is fixing.